### PR TITLE
Depend on libopenscap.so.25 explicitly

### PR DESCRIPTION
**Pull request by ekohl**
In the past we depended on the package name, but that was to support both EL7 & EL8. Now we're EL8 only so we can make it more narrow. That means we no longer need to maintain a version range, which breaks on some EL updates.

### CODE REVIEW

File: packages/plugins/rubygem-openscap/rubygem-openscap.spec [hunk 1]
```diff
@@ -4,25 +4,19 @@
...
-# require libopenscap.so.8 in an arch neutral way
-Requires: openscap >= %{min_openscap_version}
-Requires: openscap < %{max_openscap_version}
-
-BuildRequires: openscap >= %{min_openscap_version}
-BuildRequires: openscap < %{max_openscap_version}
-BuildRequires: openscap-devel
-BuildRequires: bzip2
+# Loaded via FFI
+Requires: libopenscap.so.25
```
Comment (ekohl) [inline packages/plugins/rubygem-openscap/rubygem-openscap.spec:19]: This ends up pulling in both x86_64 and i686, but I'd like to avoid making it `libopenscap.so.25()(64bit)`. So not sure how you would best solve this. Perhaps just stick to our current way, but use `< 1.4` instead.

Comment (evgeni) [reply packages/plugins/rubygem-openscap/rubygem-openscap.spec:19]: There is no guarantee whatsoever that <1.4 will keep the soname.

Can we depend on the full path? Or use a macro like _isa to get the 64 bit stuff?

Comment (ekohl) [reply packages/plugins/rubygem-openscap/rubygem-openscap.spec:19]: I hope https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/VZINPQAWEZ25JUOIW34KH7FPKEHO36XB/ will provide something useful.

Comment (ekohl) [reply packages/plugins/rubygem-openscap/rubygem-openscap.spec]: https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/7TDDGFMI7G6HOWPLDVXBLIM3DB4QDVLD/ was very quick, but very useful. I've now pushed the change without a `%check` section, but that should at least let me verify if the require bit works as expected.

Comment (ekohl) [reply packages/plugins/rubygem-openscap/rubygem-openscap.spec]: Something isn't right because this still ends up without the suffix:
```console
# rpm -q --requires rubygem-openscap-0.4.9-9.el8.noarch.rpm
libopenscap.so.25
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
ruby
ruby(release)
ruby(rubygems)
ruby(rubygems)
rubygem(ffi) >= 1.0.9
rubygem(ffi) >= 1.0.9
```

Comment (evgeni) [reply packages/plugins/rubygem-openscap/rubygem-openscap.spec]: This RPM is noarch, which resets those macros

Comment (ekohl) [reply packages/plugins/rubygem-openscap/rubygem-openscap.spec]: You're right. Updated to copy what https://src.fedoraproject.org/rpms/rubygem-ruby-vips/blob/rawhide/f/rubygem-ruby-vips.spec#_29-30 does and verified on EL8 that it does what we want: pulls in only openscap.x86_64 and not openscap.i686.

File: packages/plugins/rubygem-openscap/rubygem-openscap.spec [hunk 2]
```diff
@@ -3,26 +3,22 @@
...
-
-BuildRequires: openscap >= %{min_openscap_version}
-BuildRequires: openscap < %{max_openscap_version}
-BuildRequires: openscap-devel
-BuildRequires: bzip2
+# CI runs rpmlint on EL7
+%if 0%{?rhel} >= 8
+# Loaded via FFI
+Requires: (libopenscap.so.25()(64bit) if libc.so.6()(64bit))
+Requires: (libopenscap.so.25 if libc.so.6)
```
Comment (evgeni) [inline packages/plugins/rubygem-openscap/rubygem-openscap.spec:20]: This works fine if only one libc is installed. However, after `dnf install glibc.i686` on a x86_64 system, installing this RPM results in:

```
[root@139cecb257cd /]# dnf install http://koji.katello.org/kojifiles/work/tasks/8698/548698/rubygem-openscap-0.4.9-9.el8.noarch.rpm
Last metadata expiration check: 0:03:23 ago on Wed Mar  8 13:41:06 2023.
rubygem-openscap-0.4.9-9.el8.noarch.rpm                       87 kB/s |  26 kB     00:00    
Dependencies resolved.
=============================================================================================
 Package             Arch    Version                                     Repository     Size
=============================================================================================
Installing:
 rubygem-openscap    noarch  0.4.9-9.el8                                 @commandline   26 k
Installing dependencies:
 GConf2              i686    3.2.6-22.el8                                appstream     1.0 M
 GConf2              x86_64  3.2.6-22.el8                                appstream     1.0 M
 audit-libs          i686    3.0.7-4.el8                                 baseos        128 k
 brotli              i686    1.0.6-3.el8                                 baseos        322 k
 bzip2-libs          i686    1.0.6-26.el8                                baseos         49 k
 cyrus-sasl-lib      i686    2.1.27-6.el8_5                              baseos        129 k
 dbus-glib           i686    0.110-2.el8                                 baseos        138 k
 dbus-libs           i686    1:1.12.8-23.el8                             baseos        199 k
 elfutils-libelf     i686    0.187-4.el8                                 baseos        239 k
 expat               i686    2.2.5-9.el8                                 baseos        113 k
 glib2               i686    2.56.4-159.el8                              baseos        2.6 M
 gmp                 i686    1:6.1.2-10.el8                              baseos        314 k
 gnutls              i686    3.6.16-5.el8                                baseos        1.0 M
 keyutils-libs       i686    1.5.10-9.el8                                baseos         34 k
 krb5-libs           i686    1.18.2-21.el8                               baseos        906 k
 libacl              i686    2.2.53-1.el8                                baseos         37 k
 libattr             i686    2.4.48-3.el8                                baseos         27 k
 libblkid            i686    2.32.1-38.el8                               baseos        235 k
 libcap              i686    2.48-4.el8                                  baseos         76 k
 libcap-ng           i686    0.7.11-1.el8                                baseos         34 k
 libcom_err          i686    1.45.6-5.el8                                baseos         50 k
 libcurl             i686    7.61.1-25.el8                               baseos        330 k
 libdb               i686    5.3.28-42.el8_4                             baseos        825 k
 libffi              i686    3.1-23.el8                                  baseos         36 k
 libgcc              i686    8.5.0-15.el8                                baseos         88 k
 libgcrypt           i686    1.8.5-7.el8                                 baseos        446 k
 libgpg-error        i686    1.31-1.el8                                  baseos        247 k
 libidn2             i686    2.2.0-1.el8                                 baseos         95 k
 libmount            i686    2.32.1-38.el8                               baseos        249 k
 libnghttp2          i686    1.33.0-3.el8_2.1                            baseos         83 k
 libpsl              i686    0.20.2-6.el8                                baseos         62 k
 libselinux          i686    2.9-6.el8                                   baseos        172 k
 libsepol            i686    2.9-3.el8                                   baseos        366 k
 libssh              i686    0.9.6-3.el8                                 baseos        236 k
 libtasn1            i686    4.13-3.el8                                  baseos         79 k
 libtool-ltdl        i686    2.4.6-25.el8                                baseos         60 k
 libtool-ltdl        x86_64  2.4.6-25.el8                                baseos         58 k
 libunistring        i686    0.9.9-3.el8                                 baseos        429 k
 libuuid             i686    2.32.1-38.el8                               baseos         98 k
 libverto            i686    0.3.2-2.el8                                 baseos         25 k
 libxcrypt           i686    4.1.1-6.el8                                 baseos         74 k
 libxml2             i686    2.9.7-15.el8                                baseos        741 k
 libxslt             i686    1.1.32-6.el8                                baseos        263 k
 libxslt             x86_64  1.1.32-6.el8                                baseos        250 k
 libyaml             i686    0.1.7-5.el8                                 baseos         64 k
 libzstd             i686    1.4.4-1.el8                                 baseos        270 k
 lua-libs            i686    5.3.4-12.el8                                baseos        134 k
 lz4-libs            i686    1.8.3-3.el8_4                               baseos         67 k
 nettle              i686    3.4.1-7.el8                                 baseos        321 k
 openldap            i686    2.4.46-18.el8                               baseos        371 k
 openscap            i686    1.3.7-1.el8                                 appstream     3.7 M
 openscap            x86_64  1.3.7-1.el8                                 appstream     3.7 M
 openssl             x86_64  1:1.1.1k-7.el8                              baseos        709 k
 openssl-libs        i686    1:1.1.1k-7.el8                              baseos        1.5 M
 p11-kit             i686    0.23.22-1.el8                               baseos        323 k
 pcre                i686    8.42-6.el8                                  baseos        211 k
 pcre2               i686    10.32-3.el8                                 baseos        246 k
 polkit-libs         i686    0.115-15.el8                                baseos         80 k
 polkit-libs         x86_64  0.115-15.el8                                baseos         77 k
 popt                i686    1.18-1.el8                                  baseos         64 k
 procps-ng           x86_64  3.3.15-13.el8                               baseos        330 k
 psmisc              x86_64  23.1-5.el8                                  baseos        151 k
 rpm-libs            i686    4.14.3-23.el8                               baseos        376 k
 ruby                x86_64  2.7.6-138.module_el8.6.0+1192+2a73603d      appstream      89 k
 ruby-libs           x86_64  2.7.6-138.module_el8.6.0+1192+2a73603d      appstream     3.2 M
 rubygem-ffi         x86_64  1.15.5-1.el8                                foreman       115 k
 rubygem-irb         noarch  1.2.6-138.module_el8.6.0+1192+2a73603d      appstream     117 k
 rubygem-json        x86_64  2.3.0-138.module_el8.6.0+1192+2a73603d      appstream      93 k
 rubygem-openssl     x86_64  2.1.3-138.module_el8.6.0+1192+2a73603d      appstream     198 k
 rubygem-psych       x86_64  3.1.0-138.module_el8.6.0+1192+2a73603d      appstream      97 k
 rubygems            noarch  3.1.6-138.module_el8.6.0+1192+2a73603d      appstream     309 k
 sqlite-libs         i686    3.26.0-16.el8                               baseos        622 k
 systemd-libs        i686    239-67.el8                                  baseos        1.2 M
 trousers-lib        x86_64  0.3.15-1.el8                                baseos        168 k
 xmlsec1             i686    1.2.25-4.el8                                appstream     185 k
 xmlsec1             x86_64  1.2.25-4.el8                                appstream     191 k
 xmlsec1-openssl     i686    1.2.25-4.el8                                appstream      88 k
 xmlsec1-openssl     x86_64  1.2.25-4.el8                                appstream      94 k
 xz-libs             i686    5.2.4-4.el8                                 baseos        102 k
 zlib                i686    1.2.11-20.el8                               baseos        105 k
Installing weak dependencies:
 openssl-pkcs11      i686    0.4.10-3.el8                                baseos         70 k
 openssl-pkcs11      x86_64  0.4.10-3.el8                                baseos         66 k
 ruby-default-gems   noarch  2.7.6-138.module_el8.6.0+1192+2a73603d      appstream      74 k
 rubygem-bigdecimal  x86_64  2.0.0-138.module_el8.6.0+1192+2a73603d      appstream     101 k
 rubygem-bundler     noarch  2.2.24-138.module_el8.6.0+1192+2a73603d     appstream     445 k
 rubygem-io-console  x86_64  0.5.6-138.module_el8.6.0+1192+2a73603d      appstream      72 k
 rubygem-rdoc        noarch  6.2.1.1-138.module_el8.6.0+1192+2a73603d    appstream     455 k
 shared-mime-info    x86_64  1.9-3.el8                                   baseos        329 k
 trousers            x86_64  0.3.15-1.el8                                baseos        152 k
Downgrading:
 libgcrypt           x86_64  1.8.5-7.el8                                 baseos        463 k
 openssl-libs        x86_64  1:1.1.1k-7.el8                              baseos        1.5 M
 pcre2               x86_64  10.32-3.el8                                 baseos        247 k
 xz-libs             x86_64  5.2.4-4.el8                                 baseos         94 k

Transaction Summary
=============================================================================================
Install    90 Packages
Downgrade   4 Packages

Total size: 37 M
Total download size: 37 M
Is this ok [y/N]: 

```

TL;DR: it installs a shitload of i686 libs, but no i686 ruby and thus no actually functional 32 bit environment.

I am fine with this obstacle as the overall improvement is great, just wanted to highlight the corner here.

**Review by evgeni — APPROVED** [review]

**Comment by ekohl** [comment]
Now I remember why I had it as a draft earlier. I also wanted to add a `%check` section.

**Comment by ekohl** [comment]
I'm not actively working on it, so I'd be good with getting this in now as-is.
